lib/ImportStatement.js

```diff
@@ -149,10 +149,7 @@ export default class ImportStatement {
    * @return {Array} an array that can be used in `sort` and `uniq`
    */
   toNormalized(): Array<string> {
-    if (!this.defaultImport && !this.hasNamedImports() && this.hasSideEffects) {
-      return [this.path];
-    }
-    return [this.defaultImport || '', ...this.localNames()];
+    return [this.path];
```

Comment:
There has been plenty of discussion around sorting and grouping, and it's pretty clear that we need more flexibility: 
https://github.com/Galooshi/import-js/issues/508
https://github.com/Galooshi/import-js/issues/454
https://github.com/Galooshi/import-js/issues/448
https://github.com/Galooshi/import-js/issues/284

I think the best thing we could do is to look at what the sort-imports eslint plugin does and make things configurable the same way. 

I'm not going to argue much with what you think is the right order. What's important here is that we can either introduce a breaking change (that doesn't fix the problem) or keep the current sorting until we have a non-breaking and flexible solution. Introducing a breaking change on import sort order right now would mean introducing churn for every project that has been using import-js for a while.

Reply:
Two of those issues are things that I fixed myself - the empty line between imports being removed and the newly added imports being appended instead of prepended.
The other two issues are from people which want exactly the same sort order as I have currently implemented in this PR.
So I don't really see any conflicts of interest... Everyone wants the same thing.
Those issue links just highlight what I have told you in my previous comment.

Reply:
> I really don't see any other possible sorting as viable

Let me present you with a few possible sorting options that are all equally viable: 

- a-z on the variable name (the assignment)
- a-z on the module name (the path/package name)

For groups, any combination of the following seems equally viable: 
- built-in packages
- node_modules
- non-js files
- side-effect imports (those that have no assignment)
- relative paths
- absolute paths

The right thing here would be to make things configurable, so that you can choose any sorting and any grouping. If we make the default config produce the same sorting and grouping result as what we have today by default, I'll be happy to merge.

Reply:
First option:
> a-z on the variable name (the assignment)

This is not viable, because of I what I wrote in the first comment...
Since you haven't read it I will quote it here...

> Sorting by the imports themselves and not where they come from doesn't make any sense. This will produce many unnecessary/confusing merge conflicts. Example: instead of only removing a named export, the change will also include the movement of the imports for that file due to the sorting. The same scenario will be for appending named imports.

Second option:
> a-z on the module name (the path/package name)

This is what I have currently implemented...

For the grouping, it is very simple - exports from outside of the project, after which - local exports. I think everyone will find that as a reasonable default.

If you find any value in adding a custom sorting/grouping function that will benefit anyone (since I haven't seen one such person in the above mentioned issues by you),
then do that yourself. I have done enough work, I am happy with what I've got and I consider it to be the best default behavior for the project.

Reply:
I currently have a few projects using the current default sorting and grouping rules and I agree that this shouldn't be changed until it's configurable. Otherwise I'll need to turn it off and manually sort and group again, or deal with PR churn for the foreseeable future. I'm sure there are others using the default behaviour that we don't hear from.

Because this is an area where it's down to preference, and an existing default has been in use for a fairly long time, it does not make sense to change the default and push a different preference on those already using the default behaviour.

Now @trotzig has _not_ asked you to implement custom sorting configuration, only to keep the existing defaults so it doesn't create churn for other users.
I understand the frustration of having a PR held up when you feel that you have done a good job, but lets keep some perspective.

Also, you have done a great job here. As a user I always appreciate when someone makes an effort to improve OSS tools for the community

Reply:
The current setup doesn't make sense for the reason I have pointed out above. That is why I changed it to something that does. You simply don't want the sorting to be changed because that is what you have used in existing projects. I consider this argument invalid. Keeping wrong code because it is what people have used so far is not correct. Please give me an argument that supports the claim that the current sorting is in any way better than the one I have implemented.
If you can't, someone else will have to implement these features listed here, because I got tired of this.